index.js

```diff
@@ -72,7 +72,12 @@ Pool.prototype._create = function (cb) {
     } else {
       this.log('client connected')
       this.emit('connect', client)
-      cb(null, client)
+
+      if (this.options.connectionSetup && typeof this.options.connectionSetup === 'function') {
```

Comment:
… although it probably shouldn’t fail silently, so maybe just `if (this.options.connectionSetup !== undefined)`.